testsuite/classlibrary/TestUGen_RTAlloc.sc

```diff
@@ -70,210 +96,154 @@ TestUGen_RTAlloc : UnitTest {
-			Limiter: { Limiter.ar(DC.ar(1), dur: 0.01) },
-			Normalizer: { Normalizer.ar(DC.ar(1), dur: 0.01) },
-			KeyTrack: { KeyTrack.kr(FFT(LocalBuf(2**10), SinOsc.ar, 0.25), 0, 0) },
-			BeatTrack2: { BeatTrack2.kr(FFT(LocalBuf(2**8)), 10) },
-			PitchShift: { PitchShift.ar(SinOsc.ar, 0.01) }, // here 0.01s would be fine
-			// Warning: not really testing allocFail here, because FFT fails first
-			// however, we need the extra runtime for allocPass to work
-			Onsets: { K2A.ar(Onsets.kr(FFT(LocalBuf(2**8), Impulse.ar(1)))) }
-		);
-
-		var onlyPassUGens = (
-			// Couldn't reliably test allocFail on these:
-			IEnvGen: { IEnvGen.ar(Env([1,1]), 0) },
-			PanAz: { PanAz.ar(12, DC.ar(1)) },
-			Klank: { Klank.ar(`[100!12, nil, 1], DC.ar(1)) },
-			Klang: { Klang.ar(`[100!12, nil, 1]) },
-			Dshuf: { Demand.ar(DC.ar(1),0, Dshuf(1!12)) },
-			// hard to make these fail before LocalBuf or FFT fails
-			Loudness: { K2A.ar(Loudness.kr(FFT(LocalBuf(2**8), SinOsc.ar))) },
-			BeatTrack: { K2A.ar(BeatTrack.kr(FFT(LocalBuf(2**8)))) },
-			MFCC: { K2A.ar(MFCC.kr(FFT(LocalBuf(2**8)), 60)) }
-		);
-
-		// test fuctions:
-
-		// function to alloc LocalBuf to fill realtime memory:
-		// maxFrames = memSizeBytes / frameBytes = 1024 * 1000 / 4
-		var fillMemFunc = { LocalBuf(1024 * 1000 / 4) };
-
-		var testAllocPass = { |name, func, runtime = 0.001|
-			var cond = CondVar(), nonZeroOut = nil;
-			func.loadToFloatArray(runtime, server) { |out|
-				nonZeroOut = out.sum != 0;
-				cond.signalOne();
-			};
-			cond.waitFor(1) { nonZeroOut.notNil };
-			if (nonZeroOut.notNil) {
-				this.assert(nonZeroOut,
-					"% allocPass test should produce non-zero output".format(name)
-				);
-			} {
-				this.assert(false, "% allocPass test should complete".format(name));
-			}
-		};
-
-		var testAllocFail = { |name, func, runtime = 0.001|
-			var cond = CondVar(), zeroOut = nil;
-			func.loadToFloatArray(runtime, server) { |out|
-				zeroOut = out.sum == 0;
-				cond.signalOne();
-			};
-			cond.waitFor(1) { zeroOut.notNil };
-			if (zeroOut.notNil) {
-				this.assert(zeroOut,
-					"% allocFail test should produce zero output".format(name.asString)
-				);
-			} {
-				this.assert(false, "% allocFail test should complete".format(name.asString));
-			}
-		};
 
-		// special test for PartConv: requires buffer alloc
-		var partConvTest = {
-			var b = Buffer.loadCollection(server, 1!(2**9));
-			var testFunc = { PartConv.ar(DC.ar(1), 2**8, b) };
-			var memFill;
-			server.sync;
-			testAllocPass.value("PartConv", testFunc, 0.1);
-			memFill = fillMemFunc.play(server);
-			testAllocFail.value("PartConv", testFunc, 0.1);
-			memFill.free;
-		};
-
-		var memFill;
-
-		// RUN TESTS
-
-		ugenTests.keysValuesDo {|name, func|
-			testAllocPass.value(name.asString, func, 0.001);
-		};
+	test_allUGens_allocPass {
+
+		var memSizeKb = server.options.memSize;
+		var memSizeFloats = memSizeKb * 1024 / 4;
+
+		this.assertAllocPass("AllpassC", { AllpassC.ar(DC.ar(1), 1, 0) });
```

Comment:
although you have calculated `memSizeFloats` you don't use it and still rely on the magic $1$. Maybe we can make sure these tests pass with a variety of memSize?